pkg/proxy/proxier.go

```diff
@@ -548,12 +555,33 @@ func (proxier *Proxier) openPortal(service types.NamespacedName, info *serviceIn
 	if err != nil {
 		return err
 	}
+
+	publicIpCount := 0
 	for _, publicIP := range info.publicIP {
-		err = proxier.openOnePortal(net.ParseIP(publicIP), info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
+		ip := net.ParseIP(publicIP)
+		if ip == nil {
+			glog.V(2).Info("Skipping non-address public endpoint: %s", publicIP)
+			continue
+		}
+		publicIpCount++
+		err = proxier.openOnePortal(ip, info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
 		if err != nil {
 			return err
 		}
 	}
+
+	// We have an external load balancer, but no specific IPs.
+	// This indicates that the load balancer is acting as TCP relay, targeting the machine's IP.
+	// We also treat DNS names the same way, because DNS names are used when we have multiple
+	// addresses for the external load balancer (e.g. AWS ELB)
+	if info.externalLoadBalancer && publicIpCount == 0 {
```

Comment:
This assumption unfortunately will break when the external load balancer is created asynchronously, since public IPs are typically added during the creation of the load balancer. I don't think it'll cause problems though, so long as we properly clean it up when we receive the update to the service that includes public IPs for it.